lib/parse_date/int_from_string.rb

```diff
@@ -20,24 +18,15 @@ def self.earliest_year(date_str)
       return if date_str == '0000-00-00' # shpc collection has these useless dates
 
       # B.C. first (match longest string first)
-      return ParseDate.send(:earliest_century_bc, date_str) if date_str.match(YY_YY_CENTURY_BC_REGEX)
-      return ParseDate.send(:between_bc_earliest_year, date_str) if date_str.match(BETWEEN_Yn_AND_Yn_BC_REGEX)
-      return ParseDate.send(:year_int_for_bc, date_str) if date_str.match(YEAR_BC_REGEX)
+      bc_result = ParseDate.send(:earliest_year_bc_parsing, date_str)
+      return bc_result if bc_result
 
-      result ||= ParseDate.send(:between_earliest_year, date_str)
-      result ||= ParseDate.send(:hyphen_4digit_earliest_year, date_str)
-      result ||= ParseDate.send(:negative_first_four_digits, date_str)
-      result ||= ParseDate.send(:first_four_digits, date_str)
-      result ||= ParseDate.send(:year_from_mm_dd_yy, date_str)
-      result ||= ParseDate.send(:first_year_for_decade, date_str) # 198x or 201x
-      result ||= ParseDate.send(:first_year_for_century, date_str) # includes BC
-      result ||= ParseDate.send(:year_for_early_numeric, date_str)
-      unless result
-        # try removing brackets between digits in case we have 169[5] or [18]91
-        no_brackets = ParseDate.send(:remove_brackets, date_str)
-        return earliest_year(no_brackets) if no_brackets
-      end
-      result.to_i if result && year_int_valid?(result.to_i)
+      result = ParseDate.send(:earliest_year_parsing, date_str)
+      return result if result
+
+      # try removing brackets between digits in case we have 169[5] or [18]91
+      no_brackets = ParseDate.send(:remove_brackets, date_str)
+      earliest_year(no_brackets) if no_brackets
```

Comment:
personally, I didn't find the original cognitive load here too high.  Yes, I wrote the code, but there are clear groups of patterns.   I did get sick of rubocop complaining, tho.